include/ros1_bridge/factory.hpp

```diff
@@ -120,8 +120,10 @@ class Factory : public FactoryInterface
     callback = std::bind(
       &Factory<ROS1_T, ROS2_T>::ros2_callback, std::placeholders::_1, std::placeholders::_2,
       ros1_pub, ros1_type_name_, ros2_type_name_, node->get_logger(), ros2_pub);
+    auto rclcpp_qos = rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(qos));
```

Comment:
Maybe is better:
```cpp
rclcpp::QoS rclcpp_qos(rclcpp::QoSInitialization::from_rmw(qos));
```
And same as above, this method is used nowhere. Maybe, just update the other `create_ros2_subscriber` method to call `node->create_subscription<>()`:
https://github.com/ros2/ros1_bridge/blob/3015b901d0d6d3df4de810fa80757d180fa8297d/include/ros1_bridge/factory.hpp#L97-L108

And delete this one.

Reply:
> `rclcpp::QoS rclcpp_qos(rclcpp::QoSInitialization::from_rmw(qos));`

Yeah, I could do that, but I'll probably pass on it unless I need to touch this repository again, since it's just stylistic, the compiler should produce the same thing.